calico_kubernetes/calico_kubernetes.py

```diff
@@ -242,10 +243,24 @@ def _container_add(self, pid, interface):
             self._datastore_client.release_ips(set(ip_list))
             sys.exit(1)
 
-        # Create the veth, move into the container namespace, add the IP and
-        # set up the default routes.
-        logger.info("Creating the veth with namespace pid %s on interface name %s", pid, interface)
-        ep.mac = ep.provision_veth(netns.PidNamespace(pid), interface)
+        namespace = netns.PidNamespace(pid)
+
+        if CALICO_NETWORKING == 'true':
+            # Create the veth, move into the container namespace, add the IP and
+            # set up the default routes.
+            logger.info("Creating the veth with namespace pid %s on interface name %s", pid, interface)
+            ep.mac = ep.provision_veth(namespace, interface)
+        else:
+            try:
+                ep.name = self._get_veth(namespace, interface)
+            except:
```

Comment:
What is the exact Exception we're expecting here?

Let's try to avoid the Pokemon exception handling ;)